src/components/Bar.jsx

```diff
@@ -77,46 +79,58 @@ class Bar extends Component {
+    </h1>
+  }
+
+  renderLeft () {
+    const { t } = this.props
+    return <button className='coz-bar-burger' onClick={this.toggleDrawer} data-icon='icon-apps'>
+      <span className='coz-bar-hidden'>{t('drawer')}</span>
+    </button>
+  }
+
+  renderRight() {
     const { usageTracker, claudyOpened,
-      enableClaudy, drawerVisible, fireClaudy, displaySupport } = this.state
+      enableClaudy, fireClaudy, onDrawer, displayOnMobile, isPublic } = this.props
+    const { drawerVisible } = this.store
+    return (__TARGET__ !== 'mobile' || displayOnMobile) && !isPublic ? <div className='coz-bar-flex-container' key='nav'>
+      <Drawer visible={drawerVisible} onClose={this.toggleDrawer} onClaudy={(enableClaudy && (() => this.toggleClaudy(true))) || false} isClaudyLoading={fireClaudy} drawerListener={() => onDrawer(this.state.drawerVisible)} toggleSupport={this.toggleSupport} />
+      <Nav toggleSupport={this.toggleSupport} />
+      {enableClaudy &&
+        <Claudy
+          usageTracker={usageTracker}
+          fireClaudy={fireClaudy}
+          onToggle={() => this.toggleClaudy(false)}
+          opened={claudyOpened}
+        />
+      }
+    </div> : null
+  }
+
+  render () {
+    const { t } = this.props
+    const { fireClaudy, displaySupport } = this.state
+    const { barLeft, barRight, barCenter } = this.store
     return (
       <div className='coz-bar-container'>
-        <h1 lang={lang} className={`coz-bar-title ${replaceTitleOnMobile ? 'coz-bar-hide-sm' : ''}`}>
-          <img className='coz-bar-hide-sm' src={iconPath} width='32' />
-          {appEditor && <span className='coz-bar-hide-sm'>{appEditor} </span>}
-          <strong>{appName}</strong>
-          <sup className='coz-bar-hide-sm coz-bar-beta-status'>{t('beta')}</sup>
-        </h1>
-        <hr className='coz-sep-flex' />
-        {(__TARGET__ !== 'mobile' || displayOnMobile) && !isPublic &&
-          <div className='coz-bar-flex-container'>
-            <button className='coz-bar-burger' onClick={this.toggleDrawer} data-icon='icon-apps'>
-              <span className='coz-bar-hidden'>{t('drawer')}</span>
-            </button>
-            <Drawer visible={drawerVisible} onClose={this.toggleDrawer} onClaudy={(enableClaudy && (() => this.toggleClaudy(true))) || false} isClaudyLoading={fireClaudy} drawerListener={() => onDrawer(this.state.drawerVisible)} toggleSupport={this.toggleSupport} />
-            <Nav toggleSupport={this.toggleSupport} />
-            {enableClaudy &&
-              <Claudy
-                usageTracker={usageTracker}
-                fireClaudy={fireClaudy}
-                onToggle={() => this.toggleClaudy(false)}
-                opened={claudyOpened}
-              />
-            }
-          </div>
-        }
-        {displaySupport &&
-          <SupportModal
-            onClose={this.toggleSupport}
-          />
-        }
+        { barLeft || this.renderLeft() }
+        { barCenter || this.renderCenter() }
```

Comment:
I would prefer to use declarative components as always.

Reply:
What do you mean ? Something like : 

```js
{ barCenter || <DefaultBarCenter /> }
```

?

Reply:
Non, je veux dire comme toujours avec preact/react et jsx : 

```jsx
{ barLeft || <RenderLeft t={t} /> }
```

Reply:
Tu as lu ce que j'ai écrit ? C'est ce que je met.